test/test_fourier.jl

```diff
@@ -0,0 +1,419 @@
+
+    s_discard = Symbol.(discard)
+    test_result = true
+
+    for key in propertynames(table1)
+        if key in s_discard
+            continue
+        end
+        oe, oc = getproperty(table1,key), getproperty(table1,key)
+        if oe isa Integer || oe isa String
+            if !(oe==oc) 
+                test_result = false
+                break
+            end
+        elseif isnothing(oe)
+            if !(isnothing(oc)) 
+                test_result = false
+                break
+            end
+        else
+            if !(≈(oe,oc,rtol=rtol)) 
+                test_result = false
+                break
+            end
+        end
+    end
+
+    table1 = Metadata.drop_metadata(table1)
+    table2 = Metadata.drop_metadata(table2)
+
+    for field in names(table1)
+        if field in discard
+            continue
+        end
+        oe, oc = table1[!,field], table2[!,field]
+
+        if !(≈(oe,oc,rtol=rtol)) 
+            test_result = false
+            break
+        end
+    end
+    @test test_result
+end
+
+@testset "positive_fft_bins" begin
+    freq = fftfreq(11)
+    good = freq .> 0
+    goodbins = positive_fft_bins(11)
+    @test freq[good]==freq[goodbins]
+end
+
+@testset "test_coherence" begin
+
+    fid = h5open(joinpath(@__DIR__ ,"data","sample_variable_lc.h5"),"r")
+    data = (fid["dataset1"][1:10000]) .* 1000.0
+    close(fid)
+
+    data1 = rand.(Poisson.(data))
+    data2 = rand.(Poisson.(data))
+    ft1 = fft(data1)
+    ft2 = fft(data2)
+    dt = 0.01
+    N = length(data)
+    mean = Statistics.mean(data)
+    meanrate = mean / dt
+    freq = fftfreq(length(data), 1/dt)
+    good = (freq .> 0 .&& freq .< 0.1)
+    ft1, ft2 = ft1[good], ft2[good]
+    cross = normalize_periodograms(
+        ft1 .* conj(ft2), dt, N, mean_flux = mean, norm="abs", power_type="all")
+    pds1 = normalize_periodograms(
+        ft1 .* conj(ft1), dt, N, mean_flux = mean, norm="abs", power_type="real")
+    pds2 = normalize_periodograms(
+        ft2 .* conj(ft2), dt, N, mean_flux = mean, norm="abs", power_type="real")
+
+    p1noise = poisson_level("abs",meanrate=meanrate)
+    p2noise = poisson_level("abs",meanrate=meanrate)
+
+    @testset "test_intrinsic_coherence" begin
+        coh = estimate_intrinsic_coherence(
+            cross, pds1, pds2, p1noise, p2noise, N)
+        @test all(.≈(coh, 1, atol=0.001))
+    end
+
+    @testset "test_raw_high_coherence" begin
+        coh = raw_coherence(cross, pds1, pds2, p1noise, p2noise, N)
+        @test all(.≈(coh, 1, atol=0.001))
+    end
+
+    @testset "test_raw_low_coherence" begin
+        nbins = 2
+        C, P1, P2 = cross[1:nbins], pds1[1:nbins], pds2[1:nbins]
+        bsq = bias_term(P1, P2, p1noise, p2noise, N)
+        # must be lower than bsq!
+        low_coh_cross = rand.(Normal.(bsq.^0.5 / 10, bsq.^0.5 / 100)) .+ 0.0im
+        coh = raw_coherence(low_coh_cross, P1, P2, p1noise, p2noise, N)
+        @test all(coh .== 0)
```

Comment:
Again (3), same comment as https://github.com/matteobachetti/Stingray.jl/pull/2#discussion_r905554233
```suggestion
        @test all(iszero, coh)
```